salt/kubernetes-minion/init.sls

```diff
@@ -49,14 +49,12 @@ kubelet:
     - require:
       - pkg:    kubernetes-minion
       - file:   /etc/kubernetes/manifests
-#
-# note: br_netfilter is not available in some kernels
-#       not sure we really need it...
-#
-#      - kmod: br_netfilter
-#
-#br_netfilter:
-#  kmod.present
+  # TODO: This needs to wait for the node to register, which takes a few seconds.
+  # Salt doesn't seem to have a retry mechanism in the version were using, so I'm
+  # doing a horrible hack right now and open to suggestions for a better workaround
+  cmd.run:
+    - name: |
+        sleep 30 && kubectl --kubeconfig /etc/kubernetes/config uncordon {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}
```

Comment:
I guess the most elegant solution would be to implement a custom `beacon` that could send a _"node registered"_ event, [wait for that event](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.saltmod.html#salt.states.saltmod.wait_for_event), and then run the `kubectl uncordon`.

Otherwise maybe you can just run a `until ... do ...` command...